### PR TITLE
refactor: remove `source` and `location` in snafu display

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This pr mainly removes `source` and `location` in snafu display so that clear info is returned to client.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
part of #1854

### CODE REVIEW

**Review by waynexia — APPROVED** [review]

File: src/common/meta/src/error.rs [hunk 1]
```diff
@@ -222,29 +214,29 @@ pub enum Error {
...
 
-    #[snafu(display("Invalid catalog value, source: {}", source))]
+    #[snafu(display("Invalid catalog value"))]
     InvalidCatalogValue {
         source: common_catalog::error::Error,
         location: Location,
     },
 
-    #[snafu(display("{}", source))]
+    #[snafu(display("external error"))]
```
Comment (fengys1996) [inline src/common/meta/src/error.rs:223]:
```suggestion
    #[snafu(display("External error"))]
```

File: src/common/procedure/src/local/runner.rs [hunk 2]
```diff
@@ -942,7 +943,14 @@ mod tests {
 
         // Run the runner and execute the procedure.
         runner.run().await;
-        let err = meta.state().error().unwrap().to_string();
+        let err = meta
```
Comment (fengys1996) [inline src/common/procedure/src/local/runner.rs:946]: NIT: unwrap safe?

File: src/common/query/src/error.rs [hunk 3]
```diff
@@ -29,23 +29,20 @@ use statrs::StatsError;
 #[derive(Debug, Snafu)]
 #[snafu(visibility(pub))]
 pub enum Error {
-    #[snafu(display("Fail to execute Python UDF, source: {}", msg))]
+    #[snafu(display("Fail to execute Python UDF: {}", msg))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:32]:
```suggestion
    #[snafu(display("Failed to execute Python UDF: {}", msg))]
```

File: src/common/query/src/error.rs [hunk 4]
```diff
@@ -29,23 +29,20 @@ use statrs::StatsError;
...
         // TODO(discord9): find a way that prevent circle depend(query<-script<-query) and can use script's error type
         msg: String,
         location: Location,
     },
 
-    #[snafu(display(
-        "Fail to create temporary recordbatch when eval Python UDF, source: {}",
-        source
-    ))]
+    #[snafu(display("Fail to create temporary recordbatch when eval Python UDF"))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:39]:
```suggestion
    #[snafu(display("Failed to create temporary recordbatch when eval Python UDF"))]
```

File: src/common/query/src/error.rs [hunk 5]
```diff
@@ -29,23 +29,20 @@ use statrs::StatsError;
...
     UdfTempRecordBatch {
         location: Location,
         source: RecordbatchError,
     },
 
-    #[snafu(display("Fail to execute function, source: {}", source))]
+    #[snafu(display("Fail to execute function"))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:45]:
```suggestion
    #[snafu(display("Failed to execute function"))]
```

File: src/common/query/src/error.rs [hunk 6]
```diff
@@ -58,25 +55,25 @@ pub enum Error {
         location: Location,
     },
 
-    #[snafu(display("Fail to generate function, source: {}", source))]
+    #[snafu(display("Fail to generate function"))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:58]:
```suggestion
    #[snafu(display("Failed to generate function"))]
```

File: src/common/query/src/error.rs [hunk 7]
```diff
@@ -58,25 +55,25 @@ pub enum Error {
...
     GenerateFunction {
         source: StatsError,
         location: Location,
     },
 
-    #[snafu(display("Fail to cast scalar value into vector: {}", source))]
+    #[snafu(display("Fail to cast scalar value into vector"))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:64]:
```suggestion
    #[snafu(display("Failed to cast scalar value into vector"))]
```

File: src/common/query/src/error.rs [hunk 8]
```diff
@@ -58,25 +55,25 @@ pub enum Error {
...
     FromScalarValue {
         location: Location,
         source: DataTypeError,
     },
 
-    #[snafu(display("Fail to cast arrow array into vector: {}", source))]
+    #[snafu(display("Fail to cast arrow array into vector"))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:70]:
```suggestion
    #[snafu(display("Failed to cast arrow array into vector"))]
```

File: src/common/query/src/error.rs [hunk 9]
```diff
@@ -58,25 +55,25 @@ pub enum Error {
...
     FromArrowArray {
         location: Location,
         source: DataTypeError,
     },
 
-    #[snafu(display("Fail to cast arrow array into vector: {:?}, {}", data_type, source))]
+    #[snafu(display("Fail to cast arrow array into vector: {:?}", data_type))]
```
Comment (fengys1996) [inline src/common/query/src/error.rs:76]:
```suggestion
    #[snafu(display("Failed to cast arrow array into vector: {:?}", data_type))]
```

**Review by fengys1996 — APPROVED** [review]
LGTM

File: src/common/procedure/src/local/runner.rs (hunk 2 above)
Comment (shuiyisong) [reply src/common/procedure/src/local/runner.rs:946]: Yes. Even if the error doesn't contain any source field, `last().unwrap()` would still return last error found, which is itself.

**Pull request merged**